test/test_optim.py

```diff
@@ -2665,5 +2665,26 @@ def test_bn_update_eval_momentum(self):
 
 instantiate_parametrized_tests(TestLRScheduler)
 
+def _diff_fn(p, grad, opt_differentiable_state, opt):
+    p = p.clone()
+    p.grad = grad
+    opt.update_parameters([p])
+    opt.state.update(opt_differentiable_state)
+    opt.step()
+    return p
```

Comment:
You can return the opt_differentiable_state here as well to make sure all gradients flow properly

Reply:
done!